packages/dds/tree/src/util/typeUtils.ts

```diff
@@ -3,6 +3,20 @@
  * Licensed under the MIT License.
  */
 
+/**
+ * Convert a union of types to an intersection of those types. Useful for `TransformEvents`.
+ * @privateRemarks
+ * First an always true extends clause is used (T extends T) to distribute T into to a union of types contravariant over each member of the T union.
+ * Then the constraint on the type parameter in this new context is inferred, giving the intersection.
+ * @system
+ * @public
+ */
+export type UnionToIntersection<T> = (T extends T ? (k: T) => unknown : never) extends (
```

Comment:
nit. this file doesn't really need to be edited. if you keep your local main branch up to date you can run `git checkout main -- packages/dds/tree/src/util/typeUtils.ts` to clobber it with the version from main, kind of like a revert. you can even do it on whole sub-paths if you want to undo changes in an area.